core/indexer/processTransactions_test.go

```diff
@@ -420,17 +456,14 @@ func TestAlteredAddresses(t *testing.T) {
 	_, alteredAddresses := txProc.prepareTransactionsForDatabase(body, hdr, txPool, selfShardID)
 	require.Equal(t, len(expectedAlteredAccounts), len(alteredAddresses))
 
-	count := 0
 	for addrActual := range alteredAddresses {
-		for addrExpected := range expectedAlteredAccounts {
-			if addrActual == addrExpected {
-				count++
-				break
-			}
+		_, found := expectedAlteredAccounts[addrActual]
+		if !found {
+			assert.Fail(t, fmt.Sprintf("address %s not found", addrActual))
 		}
 	}
 
-	require.Equal(t, len(expectedAlteredAccounts), count)
+	require.Equal(t, len(expectedAlteredAccounts), len(alteredAddresses))
```

Comment:
after refactoring, this check if duplicated to line 457

Reply:
removed